backend/src/neuroagent/executor.py

```diff
@@ -0,0 +1,312 @@
+                executed = executor.run_code(code)
+        ```
+        The example assumes the plotly wheel has been manually downloaded and put it in the folder ./cached_wheels/plotly-wheel.wlh
+        """  # noqa: D300
+        with tempfile.TemporaryDirectory(prefix="pyodide_deno_") as runner_dir:
+            runner_path = os.path.join(runner_dir, "pyodide_runner.js")
+            deno_permissions = [  # Allow fetching + caching packages
+                "allow-net="
+                + ",".join(
+                    [
+                        "cdn.jsdelivr.net:443",  # allow loading pyodide packages
+                        "pypi.org:443,files.pythonhosted.org:443",  # allow pyodide install packages from PyPI
+                    ]
+                ),
+                "allow-read=./node_modules,./cached_wheels",
+                "allow-write=./node_modules",
+                "node-modules-dir=auto",
+            ]
+            # Create the JavaScript runner file
+            with open(runner_path, "w") as f:
+                f.write(
+                    self.JS_CODE.format(
+                        packages=self.additional_imports, code=json.dumps("")
+                    )
+                )  # Empty code
+
+            cmd = (
+                [self.deno_path, "run"]
+                + [f"--{perm}" for perm in deno_permissions]
+                + [runner_path]
+            )
+
+            # Start the server process
+            subprocess.run(  # nosec: B603
+                cmd,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                text=True,
+                timeout=self.timeout,
+            )
+            return self
+
+    def __exit__(
+        self,
+        exc_type: type[BaseException] | None,
+        exc_value: BaseException | None,
+        traceback: TracebackType | None,
+    ) -> Literal[False]:
+        """Exit context manager."""
+        return False
+
+    def run_code(self, code: str) -> SuccessOutput | FailureOutput:
+        """
+        Execute Python code in the Pyodide environment and return the result.
+
+        Parameters
+        ----------
+            code (`str`): Python code to execute.
+
+        Returns
+        -------
+            `SuccessOutput | FailureOutput`: Code output containing the result and logs or potential errors.
+        """
+        with tempfile.TemporaryDirectory(prefix="pyodide_deno_") as runner_dir:
+            runner_path = os.path.join(runner_dir, "pyodide_runner.js")
+            # Create the JavaScript runner file
+            with open(runner_path, "w") as f:
+                f.write(
+                    self.JS_CODE.format(
+                        packages=self.additional_imports, code=json.dumps(code)
+                    )
+                )
+
+            cmd = [self.deno_path, "run"] + self.deno_permissions + [runner_path]
+
+            # Start the server process
```

Comment:
AFAIK there is no `deno serve` anymore. If that is the case then this comment is misleading.